drivers/api/no_os_spi.c

```diff
@@ -121,12 +121,12 @@ int32_t no_os_spibus_init(const struct no_os_spi_init_param *param)
  */
 int32_t no_os_spi_remove(struct no_os_spi_desc *desc)
 {
-	// Remove SPI bus
-	no_os_spibus_remove(desc->bus->device_id);
-
 	if (!desc || !desc->platform_ops)
 		return -EINVAL;
 
+	// Remove SPI bus
+	no_os_spibus_remove(desc->bus->device_id);
```

Comment:
wondering if it is worth checking desc->bus == NULL prior to calling this function. or calling this function if bus != NULL.